src/plugins/hiddenRows/hiddenRows.js

```diff
@@ -154,30 +154,42 @@ class HiddenRows extends BasePlugin {
    */
   showRows(rows) {
     const currentHideConfig = this.getHiddenRows();
-    const isConfigValid = this.isValidConfig(rows);
+    const isValidConfig = this.isValidConfig(rows);
     let destinationHideConfig = currentHideConfig;
+    const hidingMapValues = this.#hiddenRowsMap.getValues().slice();
+    const isAnyRowShowed = rows.length > 0;
 
-    if (isConfigValid) {
-      destinationHideConfig = currentHideConfig.filter(row => rows.includes(row) === false);
+    if (isValidConfig && isAnyRowShowed) {
+      const physicalRows = rows.map(visualRow => this.hot.toPhysicalRow(visualRow));
+
+      // Preparing new values for hiding map.
+      arrayEach(physicalRows, (physicalRow) => {
+        hidingMapValues[physicalRow] = false;
+      });
+
+      // Preparing new hiding config.
+      destinationHideConfig = arrayReduce(hidingMapValues, (hiddenIndexes, isHidden, physicalIndex) => {
+        if (isHidden) {
+          hiddenIndexes.concat(this.hot.toVisualRow(physicalIndex));
```

Comment:
```suggestion
          hiddenIndexes.push(this.hot.toVisualRow(physicalIndex));
```
The same as [here](https://github.com/handsontable/handsontable/pull/7223/files#r481876518)